### PR TITLE
`yii\web\JsonParser`: convert JSONP to JSON

### PR DESCRIPTION
**Pull request by WinterSilence**
Converts JSONP to JSON parse data

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  |

### CODE REVIEW

**Comment by WinterSilence** [comment]
**pr:missing usecase**

in config:
~~~php
'request' => [
    'parsers' => [
        'application/json' => \yii\web\JsonParser::class,
       'application/javascript' => \yii\web\JsonParser::class,
     ]
]
~~~
~~~php
$rawBody = 'jQuery13476({"name":"Max","id":1234,"rank":1});';
$contentType = 'application/javascript';
$parser->parse($rawBody, $contentType);
~~~

**Comment by WinterSilence** [comment]
**pr:request for unit tests**
I'm append test for `Request::getBodyParams()` because JsonParser don't have test case.

File: framework/web/JsonParser.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
  * ```php
  * 'request' => [
  *     'parsers' => [
- *         'application/json' => 'yii\web\JsonParser',
+ *         'application/json' => yii\web\JsonParser::class,
```
Comment (machour) [inline framework/web/JsonParser.php:21]: Please rollback this change.

**Review by machour — CHANGES_REQUESTED** [review]
Add a line to CHANGELOG.md

File: framework/web/JsonParser.php (hunk 1 above)
Comment (WinterSilence) [reply framework/web/JsonParser.php:21]: @machour why you want save legacy style? it's not useful now.

Comment (machour) [reply framework/web/JsonParser.php:21]: 1. This change have absolutely nothing to do with your PR.
2. Examples are supposed to be compatible with PHP 5.4, as is the framework.

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ Yii Framework 2 Change Log
 - Enh #18726: Added `yii\helpers\Json::$prettyPrint` (rhertogh)
 - Enh #18734: Added `yii\validators\EmailValidator::$enableLocalIDN` (brandonkelly)
 - Enh #18656: Added ability for `yii serve`'s `--router` param to take an alias (markhuot)
+- Enh #18789: Added support JSONP in `yii\web\JsonParser::parse()` (WinterSilence)
```
Comment (machour) [inline framework/CHANGELOG.md:23]:
```suggestion
- Enh #18789: Added JSONP support in `yii\web\JsonParser::parse()` (WinterSilence)
```

**Review by machour — APPROVED** [review]

**Comment by samdark** [comment]
The code looks alright but I'm not sure why would you need to parse JSONP on the PHP side.

**Comment by WinterSilence** [comment]
@samdark it's need to parse external requests - often sites generate blocks by JSONP, you can't call JS callback, but you can extract wrapped JSON.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks.

**Pull request merged**